lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -82,9 +82,16 @@ static void deriveBodyRawRepresentable_raw(AbstractFunctionDecl *toRawDecl) {
 
   Type rawTy = enumDecl->getRawType();
   assert(rawTy);
+  // If the raw type is a generic parameter, unwrap it to match the type of the literals.
```

Comment:
I'm not sure this is correct. You should not depend on SubstitutedType for anything other than diagnostics...